precompiled/serde_derive/Cargo.toml

```
@@ -21,7 +21,7 @@ deserialize_in_place = []
 [lib]
 proc-macro = true
 
-[target.'cfg(not(all(target_arch = "x86_64", target_os = "linux", target_env = "gnu")))'.dependencies]
+[target.'cfg(any(serde_derive_build = "source", all(target_arch = "x86_64", target_os = "linux", target_env = "gnu")))'.dependencies]
```

```suggestion
[target.'cfg(any(serde_derive_build = "source", not(all(target_arch = "x86_64", target_os = "linux", target_env = "gnu"))))'.dependencies]
```

this must be same as `include!("lib_from_source.rs");` cfg

---

Ahhh that was it! nice catch :heart_eyes: - I was in the weeds of getting this thing into CI and got stuck in the red herring

---

Oh except this still does not work :thinking: 

```
$ rm -rf target ; cargo build --target wasm32-unknown-unknown
   Compiling whatever v0.1.0 (/home/foobar/code/whatever)
   Compiling serde_derive v1.0.183 (/home/foobar/code/serde/precompiled/serde_derive)
warning: "wasm32-unknown-unknown"
    Finished dev [unoptimized + debuginfo] target(s) in 0.32s

```

---

This is still the cross-compile error as per #2582 - it now works in host (I think)

---

It works: https://github.com/sagudev/serde/actions/runs/5916401936/job/16043642831